welkin/models/base.py

```diff
@@ -82,10 +82,10 @@ def post(self, resource, *args, **kwargs):
 
         return self
 
-    def put(self, resource, *args, **kwargs):
+    def put(self, resource, data, *args, **kwargs):
         response = self._client.put(
             resource,
-            json=self,
+            json=data,
```

Comment:
We've run into this before and considered changing it but we want to keep it as true to a PUT as possible. Is definitely frustrating that welkin uses PUT instead of PATCH in some cases but we abide by it. The way we handle this is to just have a test that shows exactly how this PUT / update should be used